internal/wasm/store.go

```diff
@@ -269,9 +230,12 @@ func (m *ModuleInstance) buildExports(exports map[string]*Export) {
 			ei = &ExportInstance{Type: exp.Type, Function: m.Functions[index]}
 			// The module instance of the host function is a fake that only includes the function and its types.
 			// We need to assign the ModuleInstance when re-exporting so that any memory defined in the target is
-			// available to the wasm.Module Memory.
+			// available to the wasm.ModuleContext Memory.
+			//
+			// TODO: shouldn't this be race if multiple modules re-exporting? maybe we should limit this to only tests.
```

Comment:
@codefromthecrypt maybe this is of your interests

Reply:
yeah though I need to think about this one more clearly tomorrow :D It looks bad though. I'll change the TODO to TODO(adrian):